processor/src/trace/mod.rs

```diff
@@ -80,12 +80,19 @@ impl ExecutionTrace {
         // create a new program info instance with the underlying kernel
         let kernel = process.kernel().clone();
         let program_info = ProgramInfo::new(program_hash.into(), kernel);
-        let (main_trace, aux_trace_hints, trace_len_summary) = finalize_trace(process, rng);
+        let (main_trace, aux_trace_builders, trace_len_summary) = finalize_trace(process, rng);
+        let trace_info = TraceInfo::new_multi_segment(
+            TRACE_WIDTH,
+            AUX_TRACE_WIDTH,
+            AUX_TRACE_RAND_ELEMENTS,
+            trace_len_summary.padded_trace_len(),
```

Comment:
Should we use `main_trace.num_rows()` here instead of `trace_len_summary.padded_trace_len()`? The values should be the same, but it would be more consistent with how we read trace length on line 152 below.

Reply:
Makes sense, changed!